### PR TITLE
use RequestMultiplePermissions

### PR DESCRIPTION
**Pull request by andrekir**
- replace `onRequestPermissionsResult` with `ActivityResultContracts.RequestMultiplePermissions()`.

### CODE REVIEW

File: app/src/main/java/co/jbear/euphony_hub/MainActivity.kt [hunk 1]
```diff
@@ -42,26 +49,4 @@ class MainActivity : ComponentActivity() {
...
-            ActivityCompat.requestPermissions(this, permissions, 0)
-        }
-    }
-
-    override fun onRequestPermissionsResult(
-        requestCode: Int,
-        permissions: Array<out String>,
-        grantResults: IntArray
-    ) {
-        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
```
Comment (yeonns) [inline app/src/main/java/co/jbear/euphony_hub/MainActivity.kt:58]: ```onRequestPermissionsResult``` function is deprecated, so thank you for replacing this function!

File: app/src/main/java/co/jbear/euphony_hub/MainActivity.kt [hunk 2]
```diff
@@ -28,9 +27,17 @@ class MainActivity : ComponentActivity() {
         Manifest.permission.WRITE_EXTERNAL_STORAGE
     )
 
+    private val requestPermissionsLauncher =
+        registerForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) { permissions ->
+            if (!permissions.entries.all { it.value }) {
+                Toast.makeText(applicationContext, "You should get permissions to use euphony-hub", Toast.LENGTH_LONG).show()
```
Comment (yeonns) [inline app/src/main/java/co/jbear/euphony_hub/MainActivity.kt:33]: Now, our app name is ```euphony-hub-collector```, so would you also change this toast message like this?:
```suggestion
                Toast.makeText(applicationContext, "You should get permissions to use euphony-hub-collector", Toast.LENGTH_LONG).show()
```

**Review by yeonns — COMMENTED** [review]
@andrekir Thank you for contributing to our repository!
With your code, permission request logic has been improved 👍 
Please check my reviews :)

File: app/src/main/java/co/jbear/euphony_hub/MainActivity.kt (hunk 2 above)
Comment (andrekir) [reply app/src/main/java/co/jbear/euphony_hub/MainActivity.kt]: fixed. :thumbsup:

**Review by yeonns — APPROVED** [review]
Look so nice!
Thank you for contribution 👍

**Review by designe — APPROVED** [review]
great refactoring! :)

**Pull request merged**